tasks/f3/f3_task.go

```diff
@@ -136,99 +128,56 @@ func (f *F3Task) tryGetF3ParticipationTicket(ctx context.Context, stillOwned fun
-		Max:    1 * time.Minute,
-		Factor: 1.5,
-	}
-
-	renewLeaseWithin := f.leaseTerm / 2
-	for stillOwned() {
-		manifest, err := f.api.F3GetManifest(ctx)
-		switch {
-		case errors.Is(err, api.ErrF3Disabled):
-			log.Errorw("Cannot await F3 participation lease expiry as F3 is disabled.", "err", err)
-			return xerrors.Errorf("awaiting F3 participation lease expiry: %w", err)
-		case err != nil:
-			if backoff.Attempt() > ParticipationCheckProgressMaxAttempts {
-				log.Errorw("Too many failures while attempting to check F3  progress. Restarting participation.", "attempts", backoff.Attempt(), "err", err)
-				return nil
-			}
-			log.Errorw("Failed to check F3 progress while awaiting lease expiry. Retrying after backoff.", "attempts", backoff.Attempt(), "backoff", backoff.Duration(), "err", err)
-			time.Sleep(backoff.Duration())
-			continue
-		case manifest == nil || manifest.NetworkName != lease.Network:
-			// If we got an unexpected manifest, or no manifest, go back to the
-			// beginning and try to get another ticket. Switching from having a manifest
-			// to having no manifest can theoretically happen if the lotus node reboots
-			// and has no static manifest.
-			return nil
-		}
-		switch progress, err := f.api.F3GetProgress(ctx); {
-		case errors.Is(err, api.ErrF3Disabled):
-			log.Errorw("Cannot await F3 participation lease expiry as F3 is disabled.", "err", err)
-			return xerrors.Errorf("awaiting F3 participation lease expiry: %w", err)
-		case err != nil:
-			if backoff.Attempt() > ParticipationCheckProgressMaxAttempts {
-				log.Errorw("Too many failures while attempting to check F3  progress. Restarting participation.", "attempts", backoff.Attempt(), "err", err)
-				return nil
+			// Successfully participated
+			if !haveLease {
+				log.Infow("Successfully acquired F3 participation lease.",
+					"issuer", lease.Issuer,
+					"not-before", lease.FromInstance,
+					"not-after", lease.ToInstance(),
+				)
+				haveLease = true
 			}
-			log.Errorw("Failed to check F3 progress while awaiting lease expiry. Retrying after backoff.", "attempts", backoff.Attempt(), "backoff", backoff.Duration(), "err", err)
-			time.Sleep(backoff.Duration())
-		case progress.ID+renewLeaseWithin >= lease.ToInstance():
-			log.Infof("F3 progressed (%d) to within %d instances of lease expiry (%d). Renewing participation.", progress.ID, renewLeaseWithin, lease.ToInstance())
-			return nil
-		default:
-			remainingInstanceLease := lease.ToInstance() - progress.ID
-			waitTime := time.Duration(remainingInstanceLease-renewLeaseWithin) * manifest.CatchUpAlignment
-			if waitTime == 0 {
-				waitTime = 100 * time.Millisecond
-			}
-			log.Debugf("F3 participation lease is valid for further %d instances. Re-checking after %s.", remainingInstanceLease, waitTime)
-			time.Sleep(waitTime)
 		}
+
+		log.Debugf("F3 participation lease is valid for further %d instances.", lease.ValidityTerm)
+		time.Sleep(time.Second * 5)
 	}
 	return ctx.Err()
```

Comment:
Not new but... using `stillOwned()` and the context like this means you'll return `nil` when exiting instead of an error.

Reply:
That's actually fine, the f3 task scheduler will see that there is no task running for that SP ID, and will just spawn a new one